osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/SmartMeterServiceTest.java

```diff
@@ -64,10 +65,28 @@ void testNonExistingSmartMeter() {
     final FunctionalException exception =
         Assertions.assertThrows(
             FunctionalException.class,
-            () -> this.smartMeterService.validateSmartMeterDoesNotExist(deviceIdentification));
+            () ->
+                this.smartMeterService.validateSmartMeterDoesNotExist(deviceIdentification, false));
     assertThat(exception.getExceptionType()).isEqualTo(FunctionalExceptionType.EXISTING_DEVICE);
   }
 
+  @Test
+  void testNonExistingSmartMeterWithOverwriteIsTrue() throws FunctionalException {
```

Comment:
This is a test with an existing smart meter. Confusing method name.

Reply:
Tests are updated.